src/dns_internal.cc

```diff
@@ -210,10 +210,10 @@ class ConfigRr : public RegisteredRunner
     void endingShutdown() override;
 };
 
-RunnerRegistrationEntry(ConfigRr);
-
 } // namespace Dns
 
+DefineRunnerRegistratorIn(Dns, ConfigRr);
```

Comment:
I had to move this and other in-namespace registrator function definitions into the global namespace. We cannot simply define the registrator function for Foo::Class inside the corresponding namespace Foo because the caller of that function -- RegisterModules() in main.cc -- would not be able to declare that registrator function: Functions cannot open namespaces (to add declarations to them). And without such declaration inside RegisterModules(), a call to that registrator function there would generate compiler "not declared in this scope" errors.

We could add a third group of macros that will open the namespace and declare registrator functions there as needed, but the corresponding code noise and code duplication are not worth the namespace purity in this case. Similarly, we could add a whole dedicated header file that main.cc will `#include` instead of self-declaring these registrator functions, but that would create even more noise! The current solution is the lesser of all these evils IMO.